src/main/java/org/terasology/logic/health/RegenComponent.java

```diff
@@ -70,14 +71,17 @@ public boolean isEmpty() {
     }
 
     public int getRegenValue() {
-        float totalValue = 0;
+        float totalValue = remainder;
         for (float value: regenValue.values()) {
             totalValue += value;
         }
-        if (totalValue < 0) {
-            totalValue = 0;
+        remainder = Math.min(1, remainder);
```

Comment:
You can _actually_ do
```java
remainder = totalValue % 1 //modulo 1
```
To get the remainder when dividing by 1.
This saves you having to floor to int and then subtract to get the return value